src/Stryker.Core/Stryker.Core/Initialisation/Buildalyzer/IAnalyzerResultExtensions.cs

```diff
@@ -151,10 +152,9 @@ private static NullableContextOptions GetNullableContextOptions(this IAnalyzerRe
 
         private static bool IsSignedAssembly(this IAnalyzerResult analyzerResult) => analyzerResult.GetPropertyOrDefault("SignAssembly", false);
 
-        private static string GetAssemblyOriginatorKeyFile(this IAnalyzerResult analyzerResult) =>
-            Path.Combine(
-                Path.GetDirectoryName(analyzerResult.ProjectFilePath),
-                analyzerResult.GetPropertyOrDefault("AssemblyOriginatorKeyFile"));
+        private static string GetAssemblyOriginatorKeyFile(this IAnalyzerResult analyzerResult) => analyzerResult.GetPropertyOrDefault("AssemblyOriginatorKeyFile") is var keyFile && keyFile is not null
```

Comment:
I'd say this is a too complex method to keep in one expression. Maybe refactor this into a function with if structure?